File: schemas/kline.py
from pydantic import BaseModel, Field, ConfigDict

class KlineTick(BaseModel):
    model_config: ConfigDict = ConfigDict(populate_by_name=True)
    start_time: int = Field(alias='t')
    close_time: int = Field(alias='T')
    symbol: str = Field(alias='s')
    open_price: float = Field(alias='o')
    high_price: float = Field(alias='h')
    low_price: float = Field(alias='l')
    close_price: float = Field(alias='c')
    volume: float = Field(alias='v')
    is_closed: bool = Field(alias='x')

File: frontend/src/types/kline.ts
export interface KlinePacket {
  exchangeId: string;
  symbol: string;
  interval: string;
  openPrice: string;
  closePrice: string;
  highPrice: string;
  lowPrice: string;
  volume: string;
  trades: number;
  isClosed: boolean;
  quoteVolume: string;
  takerBuyBaseVolume: string;
  takerBuyQuoteVolume: string;
  eventId: string;
}
